src/linear_layer.cpp

```diff
@@ -535,6 +546,9 @@ void Linear::forward(BaseHiddenStates &input_states,
     // Initialization
     int batch_size = input_states.block_size;
     this->set_cap_factor_udapte(batch_size);
+    if (temp_states.linear_states.num_w == 0) {
+        temp_states.linear_states.set_num_states_w(this->mu_w.size());
+    }
```

Comment:
we shouldn't introduce smoother code hear. See above, create a `SmootherLinear` or `SLinear` (my preference) class similar to Linear . You will then add all the smoother code here

Reply:
ok. I will do that.

Reply:
solved by creating Slinear and SLSTM layers.